### ISSUE TITLE
UI Design

### ISSUE THREAD

**Issue by jounik** [vaadin/vaadin-details#1]
A component to show more content when the user requests it, usually by clicking the panel header to reveal more content below it.

Polymer Elements has `iron-collapse` which we can utilize, but we could provide a more polished element, since currently it requires some boilerplate to trigger the opening and closing, and doesn’t provide lazy-loading content for instance, and doesn’t provide any default styles for material design.

Suggested element name: `vaadin-details` (since HTML actually has a draft spec for this element named `details`).

#### UI Design
- [x] End user feature design (what does it do, responsive behavior, etc.)
- [x] Graphical design

#### Technical design
- [ ] Technical feature design (e,g, lazy-loading of the content, etc.)
- [ ] API design (element name, light dom api, etc.)

#### Attachments on Google Drive (private):

- [End user feature design](https://docs.google.com/document/d/1DBjYcUs9wMBGCne5VNIJ1fWev5pBQTk-iFVeDyd_eHk/edit)

- [Graphical designs](https://drive.google.com/drive/folders/0BzkWbWvgmkIxU1hCeE9mMUNsV2c)

**Comment by jounik** [comment]
HTML defines (or has a draft at least) for a `<details>` element, which is the same thing as this component (and WebKit and Blink implement that, Firefox doesn’t). We should perhaps try to follow that spec.

E.g.

```
<vaadin-details>
  <summary>Comments: <span>[[numOfComments]]</span></summary>
  <vaadin-details-content path="path/to/lazy-loaded-comments.html"></vaadin-details-content>
</vaadin-details>
```

or 

```
<vaadin-details>
  <summary>More info</summary>
  <p>Static content inside the disclosure.</p>
  <p>There could be multiple elements inside here, 
     and all of those would be considered as the "content", 
     apart from the `summary` element.
</vaadin-details>
```

**Comment by jounik** [comment]
Look at the [lazy-pages](http://robdodson.github.io/lazy-pages/bower_components/lazy-pages/) element as example of lazy-loading of the content

**Comment by tuomasnironen** [comment]
Prototyped UI with "Principle" tool (http://principleformac.com/). Does not offer exporting interactive prototype nor layout formatting as in original designs.

**Comment by jounik** [comment]
That looks nice, at least as a side-viewer. So Principle doesn’t offer a way to let others try out that prototype interactively?

**Comment by tuomasnironen** [comment]
Nope, unless you have the Principle installed yourself for testing project files.

**Comment by jounik** [comment]
About lazy loading: the Polymer team is about to release an element regarding that exact functionality, so we should wait for that instead of implementing something similar ourselves at this moment.
